### PR TITLE
Add `parameters` to report_card

### LINKED ISSUE #22969: [BE] add a `parameters` field to cards, and allow it to be saved and fetched via card APIs

### PR DESCRIPTION
**Pull request by qnkhuat**
As part of the attempt to parameterize Cards, this PR adds `parameters` column to `report_card` and changes APIs so we could add/update/delete it.

With this:
- `POST /api/card` and `PUT /api/card/:id` will accept a new `parameters` param
- Any APIs that return a card now will have a new `parameters` keys

Closes https://github.com/metabase/metabase/issues/22969

### CODE REVIEW

File: frontend/src/metabase/parameters/utils/cards.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ export function getTemplateTagsForParameters(card) {
 }
 
 export function getParametersFromCard(card) {
-  if (card && card.parameters) {
+  if (card && card.parameters && card.parameters.length > 0) {
```
Comment (ariya) [inline frontend/src/metabase/parameters/utils/cards.js:64]: How about `Array.isArray(card.parameters)`?

File: src/metabase/models/card.clj [hunk 2]
```diff
@@ -231,7 +234,8 @@
   For the OSS edition, there is no implementation for this function -- it is a no-op. For Metabase Enterprise Edition,
   the implementation of this function is
   [[metabase-enterprise.sandbox.models.group-table-access-policy/update-card-check-gtaps]] and is installed by that
-  namespace."} pre-update-check-sandbox-constraints
```
Comment (qnkhuat) [inline src/metabase/models/card.clj:234]: format fix

File: src/metabase/models/card.clj [hunk 3]
```diff
@@ -289,7 +294,8 @@
                                        :embedding_params       :json
                                        :query_type             :keyword
                                        :result_metadata        ::result-metadata
-                                       :visualization_settings :visualization-settings})
+                                       :visualization_settings :visualization-settings
+                                       :parameters             :parameters-list})
```
Comment (qnkhuat) [inline src/metabase/models/card.clj:298]: we use the same type for parameters on dashboard

**Review by noahmoss — APPROVED** [review]
Looks good, just small nits. Accepting to unblock, though my understanding (from earlier) is that this won't be merged immediately.

File: src/metabase/util/schema.clj [hunk 4]
```diff
@@ -327,6 +327,12 @@
                                                     false)))
     (deferred-tru "value must be a valid JSON string.")))
 
+(def Parameter
+  "Schema for a valid Parameter."
+  (with-api-error-message {:id       NonBlankString
+                           s/Keyword s/Any}
```
[the remark this answers is not in the document]
Comment (qnkhuat) [reply src/metabase/util/schema.clj:333]: I think we could leave it a bit loose for now, I'll add them later when we have a fixed schema

**Review by daltojohnso — APPROVED** [review]

**Pull request merged**